src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/WhyCommand/WhyCommandUtility/DependencyGraphFinder.cs

```diff
@@ -0,0 +1,306 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NuGet.ProjectModel;
+
+namespace NuGet.CommandLine.XPlat.WhyCommandUtility
+{
+    internal static class DependencyGraphFinder
+    {
+        /// <summary>
+        /// Finds all dependency graphs for a given project.
+        /// </summary>
+        /// <param name="assetsFile">Assets file for the project.</param>
+        /// <param name="targetPackage">The package we want the dependency paths for.</param>
+        /// <param name="userInputFrameworks">List of target framework aliases.</param>
+        /// <returns>
+        /// Dictionary mapping target framework aliases to their respective dependency graphs.
+        /// Returns null if the project does not have a dependency on the target package.
+        /// </returns>
+        public static Dictionary<string, List<DependencyNode>?>? GetAllDependencyGraphs(
+            LockFile assetsFile,
+            string targetPackage,
+            List<string> userInputFrameworks)
+        {
+            var dependencyGraphPerFramework = new Dictionary<string, List<DependencyNode>?>(assetsFile.Targets.Count);
+            bool doesProjectHaveDependencyOnPackage = false;
+
+            // get all top-level package and project references for the project, categorized by target framework alias
+            Dictionary<string, List<string>> topLevelReferencesByFramework = GetTopLevelPackageAndProjectReferences(assetsFile, userInputFrameworks);
+
+            if (topLevelReferencesByFramework.Count > 0)
+            {
+                foreach (var (targetFrameworkAlias, topLevelReferences) in topLevelReferencesByFramework)
+                {
+                    LockFileTarget target = assetsFile.GetTarget(targetFrameworkAlias, runtimeIdentifier: null);
+
+                    if (target != null)
+                    {
+                        // get all package libraries for the framework
+                        IList<LockFileTargetLibrary> packageLibraries = target.Libraries;
+
+                        // if the project has a dependency on the target package, get the dependency graph
+                        if (packageLibraries.Any(l => l.Name == targetPackage))
```

Comment:
package ids are case insensitive.

Change to ignorecase comparer and please add a test (I've had regressions due to casing misses, that Andy had to fix: https://github.com/NuGet/NuGet.Client/commit/b49d8914861cf18287302ab311fc9d2566e06fe1).

Reply:
Fixed, and added a unit test for it too